lib/redis_client/cluster/router.rb

```diff
@@ -18,6 +18,8 @@ class Router
 
       def initialize(config, concurrent_worker, pool: nil, **kwargs)
         @config = config.dup
+        @original_config = config.dup
```

Comment:
It looks like the memory allocation of the object can be skipped if the flag is false.

```suggestion
        @original_config = config.dup if config.connect_with_original_config
```